library/Recurly/V3/Webhook/Webhook.hs

```diff
@@ -76,7 +80,7 @@ data Notification
   | CreditPaymentVoided Account.Code CreditPayment.Uuid
   | LegacyDunningEventNew Account.Code Invoice.Id Subscription.Uuid Transaction.Id
   | DunningEventNew Account.Code Invoice.Id Subscription.Uuid
-  deriving (Eq, Show)
+  deriving (Eq, Show, Generic, FromJSON, ToJSON, Data.Data)
```

Comment:
I was surprised you could derive JSON instances for varying non-unary (I don't know the proper term) sumtypes 🤯

Reply:
Is this a result of the derived `Data.Data`?

Reply:
It's from the `Generic` instance. I'm surprised to see `Data` there, actually. What's it used for?

Reply:
Its used in soley for the `Data.toConstr` function in  smurf's `WebhookProcessor.hs`. I left a comment about that in there. I'll probably end up removing it